packages/opentelemetry-types/src/trace/span.ts

```diff
@@ -0,0 +1,106 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SpanContext } from './span_context';
+import { Status } from './status';
+
+/**
+ * An interface that represents a span. A span represents a single operation
+ * within a trace. Examples of span might include remote procedure calls or a
+ * in-process function calls to sub-components. A Trace has a single, top-level
+ * "root" Span that in turn may have zero or more child Spans, which in turn
+ * may have children.
+ */
+export interface Span {
+  /**
+   * Returns the SpanContext object associated with this Span.
+   *
+   * @returns the SpanContext object associated with this Span.
+   */
+  context(): SpanContext;
+
+  // /**
+  //  * # TODO
+  //  * Returns the Tracer object used to create this Span.
+  //  * https://github.com/open-telemetry/opentelemetry-specification/issues/21
+  //  */
+  // tracer(): Tracer;
+
+  /**
+   * Sets an attribute to the span.
+   *
+   * @param key the key for this attribute.
+   * @param value the value for this attribute. If the value is a typeof object
+   *     it has to be JSON.stringify-able otherwise it will raise an exception.
+   */
+  setAttribute(key: string, value: string | number | boolean | object): void;
+
+  /**
+   * Adds an event to the Span.
+   *
+   * @param name the name of the event.
+   * @param [attributes] the attributes that will be added; these are
+   *     associated with this event.
+   */
+  addEvent(
+    name: string,
+    attributes?: { [key: string]: string | number | boolean | object }
+  ): void;
+
+  /**
+   * Adds a link to the Span.
+   *
+   * @param spanContext the context of the linked span.
+   * @param [attributes] the attributes that will be added; these are
+   *     associated with this link.
+   */
+  addLink(
+    spanContext: SpanContext,
+    attributes?: { [key: string]: string | number | boolean | object }
+  ): void;
+
+  /**
+   * Sets a status to the span. If used, this will override the default Span
+   * status. Default is 'OK'.
+   *
+   * @param status the Status to set.
+   */
+  setStatus(status: Status): void;
+
+  /**
+   * Updates the Span name.
+   *
+   * @param name the Span name.
+   */
+  updateName(name: string): void;
+
+  /**
+   * Marks the end of Span execution and sets the current time as the span's
+   * end time.
+   *
+   * Call to End of a Span MUST not have any effects on child spans. Those may
+   * still be running and can be ended later.
+   */
+  end(): void;
+
+  /**
+   * Returns the flag whether this span will be recorded.
+   *
+   * @returns true if this Span is active and recording information like events
+   * with the AddEvent operation and attributes using setAttributes.
+   */
+  isRecordingEvents(): boolean;
```

Comment:
From the description it sounds like the fact that it's recording events is only part of the state. Should this be called `isActive` instead or something generic?

Reply:
This flag is used to indicate whether events should be recorded or not. If it's set, `Events` and `Attributes`, etc. will be recorded. Otherwise, these events will be dropped. The only way to set this flag when we create the span. If not set, the implementation will provide a default.